aspnetcore/fundamentals/hosted-services/sample/Services/QueuedHostedService.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace BackgroundTasksSample.Services
+{
+    #region snippet1
+    public class QueuedHostedService : IHostedService
+    {
+        private CancellationTokenSource _shutdown = 
+            new CancellationTokenSource();
+        private Task _backgroundTask;
+        private readonly ILogger _logger;
+
+        public QueuedHostedService(IBackgroundTaskQueue taskQueue, ILoggerFactory loggerFactory)
+        {
+            TaskQueue = taskQueue;
+            _logger = loggerFactory.CreateLogger<QueuedHostedService>();
+        }
+
+        public IBackgroundTaskQueue TaskQueue { get; }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Queued Hosted Service is starting.");
+
+            _backgroundTask = Task.Run(BackgroundProceessing);
+
+            return Task.CompletedTask;
+        }
+
+        private async Task BackgroundProceessing()
+        {
+            while (!_shutdown.IsCancellationRequested)
+            {
+                var workItem = 
+                    await TaskQueue.DequeueAsync(_shutdown.Token);
+
+                try
+                {
+                    await workItem(_shutdown.Token);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "An error occurred executing the workItem.");
```

Comment:
Perhaps this is overkill, but I suggest using `nameof`:
```csharp
$"An error occurred executing the {nameof(workItem)}."
```
If the variable name ever changed, this hard-coded string may be missed. This also makes it clearer to the reader what your intent was with that log message.